### PR TITLE
Add option to enable speculative load hardening

### PR DESCRIPTION
**Pull request by achamayou**
Flag is applied to the library, to its dependencies, but not to the tests or tools.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,8 +35,26 @@ FetchContent_Declare(
...
+if (REGOCPP_SPECULATIVE_LOAD_HARDENING)
+  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    message(FATAL_ERROR "Speculative load hardening is only supported with Clang")
+  endif()
+endif()
+
+if (REGOCPP_SPECULATIVE_LOAD_HARDENING)
+  # Also build dependencies with speculative load hardening
+  # when the flag is enabled
+  set(CMAKE_CXX_FLAGS_SAVED "${CMAKE_CXX_FLAGS}")
```
Comment (matajoh) [inline CMakeLists.txt:48]: Why is this needed? Surely applying them using `target_compile_options` is sufficient?

Comment (achamayou) [reply CMakeLists.txt:48]: Would it not be necessary to invoke target_compile_options() on all the targets we transitively depend on?
